src/util/std_expr.h

```diff
@@ -4092,10 +4092,12 @@ class type_exprt:public exprt
 class constant_exprt:public exprt
 {
 public:
+  DEPRECATED("use constant_exprt(value, type) instead")
   constant_exprt():exprt(ID_constant)
   {
   }
 
+  DEPRECATED("use constant_exprt(value, type) instead")
```

Comment:
Probably shouldn't deprecate this, a case like

```
constant_exprt e(ty);
if(x)
  e.set_value("x");
else
  e.set_value("not x");
```
seems sensible?

Reply:
I'm a strong proponent of RAII and would thus write this as `constant_exprt e(x ? "x" : "not x", ty);`

Reply:
Could also be written as

```irep_idt value;

if(x)
  value="x";
else
  value="not x";

constant_exprt(value, ty);```